interview-questions.md

```diff
@@ -8,43 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
   Your answer:
+  A foreign key references the primary key.
 
   Researched answer:
-
-
+  In the example provided, the model Cohort has_many students, and the student model belongs_to the Cohort. When creating a database, rails automatically adds a primary key. In this example, the Cohort model will hold the primary key and the student model holds the foreign key because the foreign key is always assigned ot the belongs_to model. In Active Records associations class we learned that if you make a mistake and forget to assign a foreign key, you can generate a migration in order to update your database. The syntax will be $ rails generate migration <add_item_to_list>. Always remember to run $ rails db:migrate after generating a migration.
 
 2. Which RESTful routes must always be passed params? Why?
 
   Your answer:
 
   Researched answer:
-
-
+  Show, Create, Edit, Update, Destroy are RESTful routes that must always be passed params. Parameters, or params, are data that is sent from client back to the server. Params require certain information be passed. Then depending on whether the user wants to create or delete a item, the methods called will modify the correct item in the database. For example, if creating a database with blog posts, params can be used to require the ID and possibly a title and content. Strong params must be identified all the way at the bottom of the controller with the keyword "private. Anything below "private" is protected and cannot be called form anywhere else in the program. Params are typically the last methods inside the controller. One has to remember that if a route requires a param, the alias will also require a param gets passed.
```

Comment:
great answer. in the case of the restful they needed the id as a parameter to ensure action is taken upon the right data.